src/core/mantle/data/csv.test.ts

```diff
@@ -0,0 +1,310 @@
+import { expect, test, vi } from "vitest";
+
+import { fetchCSV } from "./csv";
+import * as Utils from "./utils";
+
+test("with header", async () => {
+  const fetchDataMock = vi.spyOn(Utils, "f");
+  fetchDataMock.mockImplementation(async () => {
+    return {
+      text: async () => `id,country,lat,,lng
+1,Japan,0,,1
+2,US,2,,3
+3,UK,4,,5
+`,
+    } as Response;
+  });
+
+  const features = await fetchCSV({
+    type: "csv",
+    url: "http://example.com",
+    csv: {
+      idColumn: "id",
+      latColumn: "lat",
+      lngColumn: "lng",
+      noHeader: false,
+    },
+  });
+
+  expect(features).toEqual([
+    {
+      id: "1",
+      geometry: {
+        type: "Point",
+        coordinates: [0, 1],
+      },
+      properties: {
+        country: "Japan",
+      },
+      range: undefined,
+    },
+    {
+      id: "2",
+      geometry: {
+        type: "Point",
+        coordinates: [2, 3],
+      },
+      properties: {
+        country: "US",
+      },
+      range: undefined,
+    },
+    {
+      id: "3",
+      geometry: {
+        type: "Point",
+        coordinates: [4, 5],
+      },
+      properties: {
+        country: "UK",
+      },
+      range: undefined,
+    },
+  ]);
+});
+
+test("has header but set index", async () => {
+  const fetchDataMock = vi.spyOn(Utils, "f");
+  fetchDataMock.mockImplementation(async () => {
+    return {
+      // lat has no header name
+      text: async () => `id,country,,,lng
+1,Japan,0,,1
+2,US,2,,3
+3,UK,4,,5
+`,
+    } as Response;
+  });
+
+  const features = await fetchCSV({
+    type: "csv",
+    url: "http://example.com",
+    csv: {
+      idColumn: 0,
+      latColumn: 2,
+      lngColumn: 100, // This should not fount.
+      noHeader: false,
+    },
+  });
+
+  expect(features).toEqual([
+    {
+      id: "1",
+      geometry: {
+        type: "Point",
+        coordinates: [0, undefined],
+      },
+      properties: {
+        country: "Japan",
+        lng: "1",
+      },
+      range: undefined,
+    },
+    {
+      id: "2",
+      geometry: {
+        type: "Point",
+        coordinates: [2, undefined],
+      },
+      properties: {
+        country: "US",
+        lng: "3",
+      },
+      range: undefined,
+    },
+    {
+      id: "3",
+      geometry: {
+        type: "Point",
+        coordinates: [4, undefined],
+      },
```

Comment:
Since a Point always requires both latitude and longitude, if either is missing, output as a feature with no geometry.

```ts
{
  id: "3",
  properties: {
    // ...
  }.
  range: undefined,
}
```

Reply:
I fixed some logic.
[fix: validate some parameter](https://github.com/reearth/reearth-web/pull/382/commits/67618d21d65ab6ae95a4ed99824e2ef14147ffd4)